crates/merkle-tree/src/tree.rs

```diff
@@ -71,16 +72,18 @@ use crate::storage::Storage;
 pub struct MerkleTree<H: FeltHash, const HEIGHT: usize> {
     root: Option<Rc<RefCell<InternalNode>>>,
     leaves: HashMap<BitVec<u8, Msb0>, Felt>,
-    nodes_removed: Vec<u64>,
+    nodes_removed: Vec<TrieStorageIndex>,
     _hasher: std::marker::PhantomData<H>,
     /// If enables, node hashes are verified as they are resolved. This allows
     /// testing for database corruption.
     verify_hashes: bool,
 }
 
 impl<H: FeltHash, const HEIGHT: usize> MerkleTree<H, HEIGHT> {
-    pub fn new(root: u64) -> Self {
-        let root = Some(Rc::new(RefCell::new(InternalNode::Unresolved(root))));
+    pub fn new(root: TrieStorageIndex) -> Self {
+        let root = Some(Rc::new(RefCell::new(InternalNode::Unresolved(
+            TrieStorageIndex::new(root.get()),
```

Comment:
This seems to be a redundant conversion here. `root` is already a `TrieStorageIndex`.